### DISCUSSION TITLE
Routing to "single trip view" - history.push changes url to reflect firebaseKey but doesn't update component.

### DISCUSSION THREAD

**Discussion by hmparsons-sde** [nss-evening-cohort-14/evening-client-side#284] [Capstones!] [answered]
1. Your Name: Holly Parsons
2. Describe your issue: When the user clicks on a "trip", they should be routed to a "single trip" view - think boards + pins. When I click "view trip", the view does not change, but the url updates to reflect the firebaseKey of the selected trip.
3. List any error messages that appear: None!
4. What have you googled? (Paste links to the things you have found from your google search - You must include at least 3 links)
- https://github.com/ReactTraining/react-router/issues/7209
- https://stackoverflow.com/questions/42941708/react-history-push-is-updating-url-but-not-navigating-to-it-in-browser
- https://teamtreehouse.com/community/historypush-does-not-render-new-url
5. What have you tried? Tell us everything you have tried.
- I have referenced both the "React basics" and "React pinterest" projects for a refresher on single views.
- Bothered my husband a lot.
- Asked this question in stand-ups; it seems that several people are also experiencing this!

Link to "Trip Card" with "view" button: https://github.com/hmparsons-sde/capstone/blob/trip-kids/src/Components/Cards/TripCard.js
Link to "Single Trip" view: https://github.com/hmparsons-sde/capstone/blob/trip-kids/src/views/SingleTrip.js
Link to Routes: https://github.com/hmparsons-sde/capstone/blob/trip-kids/src/helpers/Routes.js
Link to whole branch: https://github.com/hmparsons-sde/capstone/tree/trip-kids/src

**Comment by ajawashington** [comment]
@hmparsons-sde at first glance your SingleTrip route does not pass the user and based on your ERD and Route it looks like that is required. So change this [line](https://github.com/hmparsons-sde/capstone/blob/trip-kids/src/helpers/Routes.js#L58) to `
 component={() => <SingleTripView   user={user}/>}`

I am going to look deeper into it more.

**Reply by hmparsons-sde** [reply]
Still not working :/

**Comment by hmparsons-sde** [comment, marked as answer]
I didn't specify my Route as "exact." The router will go through all of our defined routes and return the FIRST match it finds. When we have multiple paths that have similar names, the "exact" param makes sure we ONLY navigate to the one that matches exactly :)
